### PR TITLE
[FIXED JENKINS-43486] Handle non-String parameters in env resolution

### PR DESCRIPTION
**Pull request by abayer**
* JENKINS issue(s):
    * [JENKINS-43486](https://issues.jenkins-ci.org/browse/JENKINS-43486)
* Description:
    * non-`String` parameters (most notably `booleanParam`) were causing `ClassCastException`s when we tried to add them to the map for `environment` resolution, so do an explicit `toString()` on them.
* Documentation changes:
    * n/a
* Users/aliases to notify:
    * @reviewbybees 
    * @rsandell

### CODE REVIEW

File: pipeline-model-definition/src/main/groovy/org/jenkinsci/plugins/pipeline/modeldefinition/model/Environment.groovy [hunk 1]
```diff
@@ -60,7 +60,11 @@ public class Environment implements Serializable {
 
     public Map<String,String> getCredsMap(CpsScript script) {
         Map<String,String> resolvedMap = new TreeMap<>()
-        EnvVars contextEnv = new EnvVars((Map<String,String>)script.getProperty("params"))
+        EnvVars contextEnv = new EnvVars()
+        ((Map<String,Object>)script.getProperty("params")).each { k, v ->
+            contextEnv.put(k, v.toString())
```
Comment (oleg-nenashev) [inline pipeline-model-definition/src/main/groovy/org/jenkinsci/plugins/pipeline/modeldefinition/model/Environment.groovy]: 🐛 will 💥 on null parameter, which seems to be possible in Groovy

Comment (abayer) [inline pipeline-model-definition/src/main/groovy/org/jenkinsci/plugins/pipeline/modeldefinition/model/Environment.groovy]: Turns out a null parameter causes other problems entirely -
```
java.lang.NullPointerException
	at org.jenkinsci.plugins.workflow.cps.persistence.IteratorHack.writeReplace(IteratorHack.java:52)
	at org.jenkinsci.plugins.workflow.cps.ParamsVariable.addValue(ParamsVariable.java:88)
	at org.jenkinsci.plugins.workflow.cps.ParamsVariable.getValue(ParamsVariable.java:75)
	at org.jenkinsci.plugins.workflow.cps.CpsScript.getProperty(CpsScript.java:121)
```
I'll add some NPE checks but that's a bug in `workflow-cps`, I think.

Comment (abayer) [inline pipeline-model-definition/src/main/groovy/org/jenkinsci/plugins/pipeline/modeldefinition/model/Environment.groovy]: [JENKINS-43511](https://issues.jenkins-ci.org/browse/JENKINS-43511) opened for that.

Comment (abayer) [inline pipeline-model-definition/src/main/groovy/org/jenkinsci/plugins/pipeline/modeldefinition/model/Environment.groovy]: And error-checking/test added.

Comment (abayer) [inline pipeline-model-definition/src/main/groovy/org/jenkinsci/plugins/pipeline/modeldefinition/model/Environment.groovy]: That was already fixed in [JENKINS-42367](https://issues.jenkins-ci.org/browse/JENKINS-42367), so I'm bumping to a newer `workflow-cps` version...

**Comment by emilegg** [comment]
Is this fixed ?

**Comment by abayer** [comment]
@oleg-nenashev Re-review?

**Pull request merged**